src/components/PreviewMode/PreviewMode.client.ts

```diff
@@ -24,6 +24,10 @@ class PreviewMode extends HTMLElement {
   #datocmsToken: string = '';
   #datocmsEnvironment: string = '';
 
+  static subscriptionKey({ query, variables }: QueryVariables) {
+    return JSON.stringify({ query, variables });
+  }
```

Comment:
Static function for reusability. I considered a fast hashing function as well because now the keys are so so very long.

Reply:
Could be a nice improvement. If it's a small change, I'd do it in this PR. Otherwise a separate one.